modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryVersionLocalServiceImpl.java

```diff
@@ -5,18 +5,69 @@
 
 package com.liferay.object.service.impl;
 
+import com.liferay.object.entry.util.ObjectEntryDTOConverterUtil;
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.model.ObjectEntryVersion;
 import com.liferay.object.service.base.ObjectEntryVersionLocalServiceBaseImpl;
 import com.liferay.portal.aop.AopService;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
 
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 /**
- * @author Marco Leo
+ * @author Feliphe Marinho
  */
 @Component(
 	property = "model.class.name=com.liferay.object.model.ObjectEntryVersion",
 	service = AopService.class
 )
 public class ObjectEntryVersionLocalServiceImpl
 	extends ObjectEntryVersionLocalServiceBaseImpl {
+
+	@Override
+	public ObjectEntryVersion addObjectEntryVersion(ObjectEntry objectEntry)
+		throws Exception {
+
+		ObjectEntryVersion objectEntryVersion =
+			objectEntryVersionPersistence.create(
+				counterLocalService.increment());
+
+		objectEntryVersion.setCompanyId(objectEntry.getCompanyId());
+
+		User user = _userLocalService.getUser(objectEntry.getUserId());
+
+		objectEntryVersion.setUserId(user.getUserId());
+		objectEntryVersion.setUserName(user.getFullName());
+
+		objectEntryVersion.setCreateDate(objectEntry.getCreateDate());
+		objectEntryVersion.setModifiedDate(objectEntry.getModifiedDate());
+		objectEntryVersion.setObjectEntryId(objectEntry.getObjectEntryId());
+
+		objectEntryVersion.setContent(
+			ObjectEntryDTOConverterUtil.toDTO(
+				_dtoConverterRegistry, _jsonFactory, objectEntry, user));
```

Comment:
I don’t think it’s necessary to break a line here. The order doesn’t matter, and each line is independent.

Reply:
Done